client.go

```diff
@@ -295,33 +304,25 @@ func (co *Conn) ReadMsgHeader(hdr *Header) ([]byte, error) {
 		err error
 	)
 
+	p = udpPool.Get().([]byte)
 	if isPacketConn(co.Conn) {
-		if co.UDPSize > MinMsgSize {
-			p = make([]byte, co.UDPSize)
-		} else {
-			p = make([]byte, MinMsgSize)
-		}
```

Comment:
This code was important. `UDPSize` can be set larger than `MinMsgSize`. You're not handling that at all.